packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/workflow-version-actions/components/UseAsDraftWorkflowVersionSingleRecordAction.tsx

```diff
@@ -1,66 +1,67 @@
 import { useWorkflowVersion } from '@/workflow/hooks/useWorkflowVersion';
 import { useWorkflowWithCurrentVersion } from '@/workflow/hooks/useWorkflowWithCurrentVersion';
 import { openOverrideWorkflowDraftConfirmationModalState } from '@/workflow/states/openOverrideWorkflowDraftConfirmationModalState';
+import { useState } from 'react';
 import { useSetRecoilState } from 'recoil';
 import { isDefined } from 'twenty-shared/utils';
 import { useNavigateApp } from '~/hooks/useNavigateApp';
 
-export const useUseAsDraftWorkflowVersionSingleRecordAction: ActionHookWithoutObjectMetadataItem =
-  () => {
-    const recordId = useSelectedRecordIdOrThrow();
+export const UseAsDraftWorkflowVersionSingleRecordAction = () => {
+  const recordId = useSelectedRecordIdOrThrow();
+  const workflowVersion = useWorkflowVersion(recordId);
+  const workflow = useWorkflowWithCurrentVersion(
+    workflowVersion?.workflow?.id ?? '',
+  );
+  const { createDraftFromWorkflowVersion } =
+    useCreateDraftFromWorkflowVersion();
+  const setOpenOverrideWorkflowDraftConfirmationModal = useSetRecoilState(
+    openOverrideWorkflowDraftConfirmationModalState,
+  );
+  const navigate = useNavigateApp();
+  const [hasNavigated, setHasNavigated] = useState(false);
 
-    const workflowVersion = useWorkflowVersion(recordId);
+  const hasAlreadyDraftVersion =
+    workflow?.versions.some((version) => version.status === 'DRAFT') || false;
 
-    const workflow = useWorkflowWithCurrentVersion(
-      workflowVersion?.workflow?.id ?? '',
-    );
+  const onClick = () => {
+    if (!isDefined(workflowVersion) || !isDefined(workflow) || hasNavigated) {
+      return;
+    }
 
-    const { createDraftFromWorkflowVersion } =
-      useCreateDraftFromWorkflowVersion();
-
-    const setOpenOverrideWorkflowDraftConfirmationModal = useSetRecoilState(
-      openOverrideWorkflowDraftConfirmationModalState,
-    );
-
-    const navigate = useNavigateApp();
-
-    const hasAlreadyDraftVersion =
-      workflow?.versions.some((version) => version.status === 'DRAFT') || false;
-
-    const onClick = async () => {
-      if (!isDefined(workflowVersion) || !isDefined(workflow)) {
-        return;
-      }
-
-      if (hasAlreadyDraftVersion) {
-        setOpenOverrideWorkflowDraftConfirmationModal(true);
-      } else {
+    if (hasAlreadyDraftVersion) {
+      setOpenOverrideWorkflowDraftConfirmationModal(true);
+    } else {
+      const executeAction = async () => {
         await createDraftFromWorkflowVersion({
           workflowId: workflowVersion.workflow.id,
           workflowVersionIdToCopy: workflowVersion.id,
         });
+
         navigate(AppPath.RecordShowPage, {
           objectNameSingular: CoreObjectNameSingular.Workflow,
           objectRecordId: workflowVersion.workflow.id,
         });
-      }
-    };
 
-    const ConfirmationModal = isDefined(workflowVersion) ? (
+        setHasNavigated(true);
+      };
+
+      executeAction();
```

Comment:
Could we add WithoutWaiting in the naming because it's not obvious at first sight.